### ISSUE TITLE
Export all XICs?

### ISSUE THREAD

**Issue by phlaaj** [mzmine/mzmine#1542]
### Basic information

* My operating system: macOS 13.6.1
* My MZmine version: 3.9.0

### What happened

I would like to export all XIC to research alternate deconvolution algorithms (potentially to be contributed to MZMine eventually): I don't love how the current deconvolution algorithms handle noisy features, so would like to play around with some alternate ideas. But to do this, I'd like to access the initial XIC (before feature deconvolution).

I know it's possible to export feature XICs manually one at a time: Feature Table, by right clicking a feature row > Show > XIC (quick or dialog), and then from the TIC/XIC visualizer, right click > Export data... > to Excel.

I also know it's possible to select multiple features to be visualized (image attached), however typically my feature tables contain >5000 rows, so ticking each box manually one at a time is impractical. Adding numerous features also really slows down/freezes the visualizer, so ideally looking for a way to export without having to visualize first.

What I'd like to do is just export all XICs (chromatogram data) from all features in a feature table at once (i.e., simultaneously in a single export operation), and ideally without having to go through the visualizer first (which sometimes hangs my computer). Do any of the export modules/file formats already do this? from what I can see, the .csv export includes everything in the feature table other than the XIC traces 😢 How to get the XICs exported from MZMine?
![Screenshot 2023-11-22 at 11 29 52 AM](https://github.com/mzmine/mzmine3/assets/115373239/901956ef-5048-4148-9d4f-6d27b9371857)

**Comment by SteffenHeu** [comment]
Hi, you can use the FeatureML export, which should include the XIC data. I think it can be accessed by open ms, so there might be a parser for it in pyOpenMS.

**Comment by phlaaj** [comment]
Thank you for the fast reply @SteffenHeu ! Unfortunately doesn't look like the FeatureML file contains the XICs.

Any other methods in MZMine to export XICs?

![Screenshot 2023-11-22 at 12 43 38 PM](https://github.com/mzmine/mzmine3/assets/115373239/2a5c2290-de15-4263-a579-4ef1ddb172c7)

**Comment by SteffenHeu** [comment]
you are right, the output and the code generating the output looks kind of random.

![grafik](https://github.com/mzmine/mzmine3/assets/37407705/fd911da5-ae49-44d2-a2a1-9d7e88550bc4)

@christophuv Can you confirm this is the expected behaviour? Should this not contain the m/z and intensity values at each rt?

**Comment by SteffenHeu** [comment]
Other than that, we don't currently have an export that contains all the xics. 

Unfortunately, I'm currently busy writing my thesis. However, i think what you need can be programmed relatively simple. 
- from the io.github.mzmine.modules.example package, copy FeatureListRowLearnerTask, LeanerParameters, and LearnerModule to a new package
- add the logic for the export
- Line 127 accesses the feature object
- feature.getFeatureData() gives you the xics, from there you can get everything you need
- write it to a file
- add your module class to the MainMenu.xml and make the Module.runModule create a task

**Comment by christophuv** [comment]
The featureML files are not supposed to contain CIC (at least I do not know that they should) as the format is feature-centric. 
In its current form the featureML module only generates the featureML file in a way that allows ToppVIEW to generate an overlay in the raw data (i.e., small rectangles that indicate the min/max of mz/RT of detected features.

**Comment by SteffenHeu** [comment]
ok, thank you for the calrification 👍 then i misunderstood the file format

**Comment by SteffenHeu** [comment]
This is now done with the https://mzmine.github.io/mzmine_documentation/module_docs/io_export_feature_data/export-feature-data.html export
